numba_dpex/core/kernel_interface/indexers.py

```diff
@@ -74,6 +81,50 @@ def size(self):
         else:
             return self[0]
 
+    @property
+    def ndim(self) -> int:
+        """Returns the rank of a Range object.
+
+        Returns:
+            int: Number of dimensions in the Range object
+        """
+        return len(self)
+
+    @property
+    def dim0(self) -> int:
+        """Return the extent of the first dimension for the Range object.
+
+        Returns:
+            int: Extent of first dimension for the Range object
+        """
+        return self[0]
+
+    @property
+    def dim1(self) -> int:
+        """Return the extent of the second dimension for the Range object.
+
+        Returns:
+            int: Extent of second dimension for the Range object or -1 for 1D
+            Range
+        """
+        try:
+            return self[1]
+        except:
```

Comment:
```suggestion
        except IndexError:
```

Reply:
Addressed in af0208dde9d6763117de8789a5924243d60e910e